pkg/pillar/cmd/volumemgr/sizemgmt.go

```diff
@@ -44,23 +44,26 @@ func getRemainingDiskSpace(ctxPtr *volumemgrContext) (uint64, error) {
 				iterVolumeStatus.Key(), iterVolumeStatus.State)
 			continue
 		}
-		cfg := lookupVolumeConfig(ctxPtr, iterVolumeStatus.Key())
 		sizeToUseInCalculation := uint64(iterVolumeStatus.CurrentSize)
+		hasNoAppReferences := false
+		cfg := lookupVolumeConfig(ctxPtr, iterVolumeStatus.Key())
 		if cfg == nil {
-			// we have no config with this volume, so it will be purged
-			log.Noticef("getRemainingDiskSpace: Volume %s not found in VolumeConfigs, ignore",
+			// we have no config with this volume, so it cannot have app references
+			log.Noticef("getRemainingDiskSpace: Volume %s not found in VolumeConfigs, assume no app references",
 				iterVolumeStatus.Key())
-			continue
+			hasNoAppReferences = true
+		} else {
+			hasNoAppReferences = cfg.HasNoAppReferences
 		}
 		if vault.ReadPersistType() == types.PersistZFS {
```

Comment:
Why are we checking ZFS first? This means that even if no app references we end up using the max and not current, but it can't grow to the max since no app.

If there is a reason for this based on how we reserve space with ZFS then it would make sense to add a comment.

Reply:
For zfs we allocate the whole space of volume even without references. Comment added.